From 77fd06e7289ea42513eab3f3781fbfe4daa13a63 Mon Sep 17 00:00:00 2001
From: "Mateusz [Jazn] Niemojewski" <mojes@wp.pl>
Date: Fri, 11 Aug 2023 22:21:18 +0200
Subject: [PATCH] Signed-off-by: Mateusz [Jazn] Niemojewski <mojes@wp.pl>

Add OVP functionality to lm3630a backlight driver. Setting OVP to 32V.
---
 drivers/video/backlight/lm3630a_bl.c     | 3 +++
 include/linux/platform_data/lm3630a_bl.h | 1 +
 2 files changed, 4 insertions(+)

diff --git a/drivers/video/backlight/lm3630a_bl.c b/drivers/video/backlight/lm3630a_bl.c
index 1dbf3d0e9f..e204ef21c6 100644
--- a/drivers/video/backlight/lm3630a_bl.c
+++ b/drivers/video/backlight/lm3630a_bl.c
@@ -102,6 +102,9 @@ static int lm3630a_chip_init(struct lm3630a_chip *pchip)
 	// clear OCP (overcurrent bits)
 	value &= ~(1 << 3);
 	value &= ~(1 << 4);
+	// setup OVP (overvoltage bits) 32V - is it safe?
+	value &= ~(1 << 5);
+	value |=  (1 << 6);
 
 	rval |= lm3630a_write(pchip, REG_BOOST, value);
 
diff --git a/include/linux/platform_data/lm3630a_bl.h b/include/linux/platform_data/lm3630a_bl.h
index 7a8155e4e3..54b23bc6eb 100644
--- a/include/linux/platform_data/lm3630a_bl.h
+++ b/include/linux/platform_data/lm3630a_bl.h
@@ -36,6 +36,7 @@ enum lm3630a_ledb_ctrl {
 #define LM3630A_BOOST_USE_1MHZ BIT(0)
 #define LM3630A_BOOST_SHIFT BIT(1)
 #define LM3630A_BOOST_OCP BIT(3) & Bit(4)
+#define LM3630A_BOOST_OVP BIT(5) & Bit(6)
 /*
  *@leda_label    : optional led a label.
  *@leda_init_brt : led a init brightness. 4~255
-- 
2.41.0

